apps/web/components/apps/installation/ConfigureStepCard.tsx

```diff
@@ -0,0 +1,211 @@
+  ...props
+}) => {
+  const { control, getValues } = useFormContext<TEventTypesForm>();
+  const { fields, update } = useFieldArray({
+    control,
+    name: "eventTypes",
+    keyName: "fieldId",
+  });
+
+  const submitRefs = useRef<Array<React.RefObject<HTMLButtonElement>>>([]);
+  submitRefs.current = fields.map(
+    (_ref, index) => (submitRefs.current[index] = React.createRef<HTMLButtonElement>())
+  );
+  const mainForSubmitRef = useRef<HTMLButtonElement>(null);
+  const [updatedEventTypesStatus, setUpdatedEventTypesStatus] = useState(
+    fields.filter((field) => field.selected).map((field) => ({ id: field.id, updated: false }))
+  );
+  const [submit, setSubmit] = useState(false);
+  const allUpdated = updatedEventTypesStatus.every((item) => item.updated);
+
+  useEffect(() => {
+    setUpdatedEventTypesStatus((prev) =>
+      prev.filter((state) => fields.some((field) => field.id === state.id && field.selected))
+    );
+    if (!fields.some((field) => field.selected)) {
+      setConfigureStep(false);
+    }
+  }, [fields]);
+
+  useEffect(() => {
+    if (submit && allUpdated && mainForSubmitRef.current) {
+      mainForSubmitRef.current?.click();
+      setSubmit(false);
+    }
+  }, [submit, allUpdated, getValues, mainForSubmitRef]);
+
+  return (
+    formPortalRef?.current &&
+    createPortal(
+      <div className="mt-8">
+        <div className="flex flex-col space-y-6">
+          {fields.map((field, index) => {
+            return (
+              field.selected && (
+                <EventTypeAppSettingsForm
+                  key={field.fieldId}
+                  eventType={field}
+                  handleDelete={() => {
+                    const eventMetadataDb = eventTypes?.find(
+                      (eventType) => eventType.id == field.id
+                    )?.metadata;
+                    update(index, { ...field, selected: false, metadata: eventMetadataDb });
+                  }}
+                  onSubmit={(data) => {
+                    update(index, { ...field, metadata: data });
+                    setUpdatedEventTypesStatus((prev) => {
+                      const temp = prev.map((item) =>
+                        item.id === field.id ? { ...item, updated: true } : item
+                      );
+                      return temp;
+                    });
+                  }}
+                  ref={submitRefs.current[index]}
+                  {...props}
+                />
+              )
+            );
+          })}
+        </div>
+
+        <button form="outer-event-type-form" type="submit" className="hidden" ref={mainForSubmitRef}>
+          Save
+        </button>
+        <Button
+          className="text-md mt-6 w-full justify-center"
+          type="button"
+          onClick={(e) => {
+            submitRefs.current.reverse().map((ref) => ref.current?.click());
+            setSubmit(true);
+          }}
+          loading={loading}>
+          Save
+        </Button>
```

Comment:
Can you explain what these two buttons are for? Why is one of them hidden?

Reply:
here in this scenario we have nested forms, and we want to submit the outer form only when all the inner forms have been filled out.
in this case these are the 2 individual inner forms
![Screenshot 2024-04-03 at 6 17 09 AM](https://github.com/calcom/cal.com/assets/32706411/038a9c56-46ee-41a3-9a4f-55e2a6172dd8)

`submitRefs.current.reverse().map((ref) => ref.current?.click());` is iterating over the inner forms and triggering their submission. However, we want to ensure that the outer form is only submitted when all the inner forms contain values. To achieve this, you need to keep track of whether all the inner forms are filled out. this is what `updatedEventTypesStatus` state is for.

the hidden button is the actual submit button for the outer form which we trigger when all the inner form values have been filled out and saved.